src/Workspaces/Core/MSBuild/MSBuild/ProjectFile/ProjectFile.cs

```diff
@@ -161,6 +161,12 @@ private ProjectFileInfo CreateProjectFileInfo(MSB.Execution.ProjectInstance proj
                 .Select(MakeNonSourceFileDocumentFileInfo)
                 .ToImmutableArray();
 
+            var projectCapabilities = project.GetItems(ItemNames.ProjectCapability).SelectAsArray(item => item.ToString());
+            var contentFileInfo = GetContentFiles(project);
+            var isSdkStyle = IsSdkStyleProject(loadedProject);
+            var outputKind = project.ReadPropertyString(PropertyNames.OutputType);
+            var projectTelemetryMetadata = new ProjectTelemetryMetadata(projectCapabilities, contentFileInfo, isSdkStyle);
```

Comment:
❗ This computation should not be performed when telemetry is disabled.

Reply:
Unfortunately I don't have that information at this layer - however this is essentially reading a few msbuild properties so I don't think should be that impactful.  

In the next push I've also added a check to prevent the rest of the processing (e.g. hashing and reporting of the actual event) based on the telemetry level.